api/server/router/image/image_routes.go

```diff
@@ -226,18 +226,21 @@ func (s *imageRouter) getImagesJSON(ctx context.Context, w http.ResponseWriter,
 	if err != nil {
 		return err
 	}
-
-	version := httputils.VersionFromContext(ctx)
-	if versions.LessThan(version, "1.41") {
+	var sharedSize bool
+	if versions.LessThan(httputils.VersionFromContext(ctx), "1.41") {
 		filterParam := r.Form.Get("filter")
 		if filterParam != "" {
 			imageFilters.Add("reference", filterParam)
 		}
+	} else {
+		// NOTE: Support for the "shared-size" parameter was added in API 1.42.
```

Comment:
nit: _slightly_ leaning toward making this separate from the check above, i.e.

```suggestion
	}

	if versions.GreaterThanOrEqualTo(httputils.VersionFromContext(ctx), "1.42") {
		// Support for the "shared-size" parameter was added in API 1.42.	
```